drivers/tee/tee_shm.c

```diff
@@ -183,68 +223,66 @@ struct tee_shm *tee_shm_alloc_user_buf(struct tee_context *ctx, size_t size)
  */
 struct tee_shm *tee_shm_alloc_kernel_buf(struct tee_context *ctx, size_t size)
 {
-	u32 flags = TEE_SHM_DYNAMIC | TEE_SHM_POOL;
+	u32 flags = TEE_SHM_DMA_BUF | TEE_SHM_POOL;
 
-	return shm_alloc_helper(ctx, size, PAGE_SIZE, flags, -1);
+	return tee_shm_alloc(ctx, size, flags);
 }
 EXPORT_SYMBOL_GPL(tee_shm_alloc_kernel_buf);
 
-/**
- * tee_shm_alloc_priv_buf() - Allocate shared memory for a privately shared
- *			      kernel buffer
- * @ctx:	Context that allocates the shared memory
- * @size:	Requested size of shared memory
- *
- * This function returns similar shared memory as
- * tee_shm_alloc_kernel_buf(), but with the difference that the memory
- * might not be registered in secure world in case the driver supports
- * passing memory not registered in advance.
- *
- * This function should normally only be used internally in the TEE
- * drivers.
- *
- * @returns a pointer to 'struct tee_shm'
- */
-struct tee_shm *tee_shm_alloc_priv_buf(struct tee_context *ctx, size_t size)
+static int shm_get_kernel_pages(unsigned long start, size_t page_count,
```

Comment:
Please put shm_get_kernel_pages to another new patch. Let's do only code replace in this patch. Thus we can track code change better.